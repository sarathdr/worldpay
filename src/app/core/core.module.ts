import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
