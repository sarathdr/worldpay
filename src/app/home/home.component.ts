import { Component, Input, OnInit } from '@angular/core';
import { Cart } from '../core/models/cart.interface';

@Component({
  selector: 'home',
  template: `
    <div class="cart">

      <table mat-table [dataSource]="cart.products" class="mat-elevation-z8">

        <ng-container matColumnDef="name">
          <th mat-header-cell *matHeaderCellDef> Products</th>
          <td mat-cell *matCellDef="let element"> {{ element.name }} </td>
          <td mat-footer-cell *matFooterCellDef> Total</td>
        </ng-container>

        <ng-container matColumnDef="price">
          <th mat-header-cell *matHeaderCellDef> Price</th>
          <td mat-cell *matCellDef="let element"> {{ element.price | currency: cart.currency }} </td>
          <td mat-footer-cell *matFooterCellDef> <b>{{ cart.total | currency: cart.currency }}</b></td>

        </ng-container>

        <tr mat-header-row *matHeaderRowDef="['name', 'price']"></tr>
        <tr mat-row *matRowDef="let row; columns: ['name', 'price'];"></tr>
        <tr mat-footer-row *matFooterRowDef="['name', 'price']; sticky: true"></tr>
      </table>
      <button mat-raised-button color="primary" routerLink="/payment">Continue to payment</button>
    </div>
  `,
  styles: [
      `
    .cart {
        margin: 0 auto;
        display: flex;
        flex-direction: column;
        width: 400px;
    }
    .cart table {
        width: 400px;
        margin-bottom: 30px;
    }
    `
  ]
})
export class HomeComponent {
  @Input() cart: Cart;
}
