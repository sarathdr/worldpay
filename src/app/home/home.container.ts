import { Component } from '@angular/core';
import { Observable } from 'rxjs';
import { CartService } from '../core/services/cart.service';
import { Cart } from '../core/models/cart.interface';

@Component({
  selector: 'home-page',
  template: `
    <home
        [cart]="cart$ | async">
    </home>
  `,
})
export class HomeContainer {
  cart$: Observable<Cart>;

  constructor(cartService: CartService) {
    this.cart$ = cartService.getCart();
  }

}
