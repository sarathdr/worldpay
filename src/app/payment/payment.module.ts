import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthorizeComponent } from './authorize/authorize.component';
import { AuthorizeContainer } from './authorize/authorize.container';
import { ConfirmComponent } from './confirm/confirm.component';
import { ConfirmContainer } from './confirm/confirm.container';
import { RefundComponent } from './refund/refund.component';
import { RefundContainer } from './refund/refund.container';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { MatButtonModule, MatFormFieldModule, MatInputModule, MatSelectModule } from '@angular/material';
import { AuthorizeFormComponent } from './authorize/authorize-form.component';
import { ConfirmPaymentGuard } from './guards/confirm-payment.guard';
import { ExpiryDateModule } from '../core/directives/expiry-date.module';
import { RefundPaymentGuard } from './guards/refund-payment.guard';
import { RefundSuccessComponent } from './refund/refund-success.component';

@NgModule({
  declarations: [
    AuthorizeComponent,
    AuthorizeFormComponent,
    AuthorizeContainer,
    ConfirmComponent,
    ConfirmContainer,
    RefundComponent,
    RefundSuccessComponent,
    RefundContainer
  ],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule.forChild([
      {
        path: '',
        component: AuthorizeContainer
      },
      {
        path: 'confirm',
        component: ConfirmContainer,
        canActivate:[ConfirmPaymentGuard],
        canDeactivate:[ConfirmPaymentGuard]
      },
      {
        path: 'refund',
        component: RefundContainer,
        canActivate:[RefundPaymentGuard],
        canDeactivate:[RefundPaymentGuard]
      }
    ]),
    MatFormFieldModule,
    MatInputModule,
    MatSelectModule,
    ReactiveFormsModule,
    MatButtonModule,
    ExpiryDateModule
  ],
  providers:[
    ConfirmPaymentGuard,
    RefundPaymentGuard
  ]
})
export class PaymentModule { }
