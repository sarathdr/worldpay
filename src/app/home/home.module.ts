import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HomeComponent } from './home.component';
import { RouterModule } from '@angular/router';
import { HomeContainer } from './home.container';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';


@NgModule({
  declarations: [HomeContainer, HomeComponent],
  imports: [
    CommonModule,
    RouterModule.forChild([
      {
        path: '',
        component: HomeContainer
      }
    ]),
    MatListModule,
    MatButtonModule,
    MatTableModule
  ]
})
export class HomeModule {
}
