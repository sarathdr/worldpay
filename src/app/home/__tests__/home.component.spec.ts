import { TestBed } from '@angular/core/testing';
import { HomeComponent } from '../home.component';
import { Currency } from '../../core/models/payment.interface';
import { Component } from '@angular/core';
import { MatButtonModule, MatListModule, MatTableModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';

describe('HomeComponent', () => {

  let fixture;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        MatListModule,
        MatButtonModule,
        MatTableModule
      ],
      declarations: [
        HomeComponent,
        TestHomeComponent
      ],
    });

    fixture = TestBed.createComponent(TestHomeComponent);
  });

  it('should render the component correctly', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  @Component({
    selector: 'test-home-component',
    template: `
      <home [cart]="cart"></home>`
  })
  class TestHomeComponent {
    cart = {
      products: [{
        name: 'Pen',
        price: 10
      },
        {
          name: 'Pencil',
          price: 12
        }],
      total: 22,
      currency: Currency.GBP
    };
  }

});
