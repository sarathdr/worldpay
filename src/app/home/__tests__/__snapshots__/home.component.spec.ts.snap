// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render the component correctly 1`] = `
<test-home-component
  cart={[Function Object]}
>
  <home
    ng-reflect-cart="[object Object]"
  >
    <div
      class="cart"
    >
      <table
        class="mat-elevation-z8 mat-table"
        mat-table=""
        ng-reflect-data-source="[object Object],[object Object"
        role="grid"
      >
        <thead
          role="rowgroup"
        >
          
          <tr
            class="mat-header-row"
            mat-header-row=""
            role="row"
          >
            
            <th
              class="mat-header-cell cdk-column-name mat-column-name"
              mat-header-cell=""
              role="columnheader"
            >
               Products
            </th>
            <th
              class="mat-header-cell cdk-column-price mat-column-price"
              mat-header-cell=""
              role="columnheader"
            >
               Price
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
               Pen 
            </td>
            <td
              class="mat-cell cdk-column-price mat-column-price"
              mat-cell=""
              role="gridcell"
            >
               £10.00 
            </td>
          </tr>
          <tr
            class="mat-row"
            mat-row=""
            role="row"
          >
            
            <td
              class="mat-cell cdk-column-name mat-column-name"
              mat-cell=""
              role="gridcell"
            >
               Pencil 
            </td>
            <td
              class="mat-cell cdk-column-price mat-column-price"
              mat-cell=""
              role="gridcell"
            >
               £12.00 
            </td>
          </tr>
        </tbody>
        <tfoot
          class="mat-table-sticky"
          role="rowgroup"
          style="bottom: 0px; position: sticky; z-index: 10;"
        >
          
          <tr
            class="mat-footer-row"
            mat-footer-row=""
            role="row"
          >
            
            <td
              class="mat-footer-cell cdk-column-name mat-column-name mat-table-sticky"
              mat-footer-cell=""
              role="gridcell"
              style="bottom: 0px; position: sticky; z-index: 10;"
            >
               Total
            </td>
            <td
              class="mat-footer-cell cdk-column-price mat-column-price mat-table-sticky"
              mat-footer-cell=""
              role="gridcell"
              style="bottom: 0px; position: sticky; z-index: 10;"
            >
              <b>
                £22.00
              </b>
            </td>
          </tr>
        </tfoot>
      </table>
      <button
        class="mat-raised-button mat-button-base mat-primary"
        color="primary"
        mat-raised-button=""
        ng-reflect-color="primary"
        ng-reflect-router-link="/payment"
        routerlink="/payment"
        tabindex="0"
      >
        <span
          class="mat-button-wrapper"
        >
          Continue to payment
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </home>
</test-home-component>
`;
